transport/rawsocketpeer.go

```diff
@@ -64,7 +64,7 @@ func ConnectRawSocketPeer(network, address string, serialization serialize.Seria
 		return nil, err
 	}
 
-	conn, err := net.Dial(network, address)
+	conn, err := net.DialTimeout(network, address, 5 * time.Second)
```

Comment:
The approach that I would prefer for this is for the caller to pass in a context. That way the caller can set whatever timeout/deadline they want, or call the cancel function stop waiting.  However, that probably requires an API change so need to consider carefully

Reply:
See comment below

Reply:
Yes, like your comment below.  I think that would work well.

Reply:
OK great, will update the PR